src/drug_discovery/docking.py

```diff
@@ -98,38 +87,193 @@ def show_results(self):
         """
+        jobs = self._get_jobs()
+        row_ids = [job.attributes.userOutputs.results_sdf.rowId for job in jobs]
+
+        df = pd.DataFrame(
+            api.get_dataframe(
+                "Docking",
+                return_type="dict",
+                use_file_names=False,
+            )
+        )
+
+        df = df[df["ID"].isin(row_ids)]
+        file_ids = df[column_name].tolist()
+
+        existing_files = os.listdir(utils.DATA_DIRS["Docking"])
+        existing_files = [
+            "_file:" + file.replace(file_extension, "") for file in existing_files
+        ]
+        missing_files = list(set(file_ids) - set(existing_files))
+        if len(missing_files) > 0:
+            api.download_files(
+                file_ids=missing_files,
+                use_file_names=False,
+                save_to_dir=utils.DATA_DIRS["Docking"],
+            )
+
+        file_paths = [
+            os.path.join(utils.DATA_DIRS["Docking"], file_id.replace("_file:", ""))
+            for file_id in file_ids
+        ]
+
+        file_paths = ensure_file_extension(
+            file_paths=file_paths, extension=file_extension
+        )
+
+        return file_paths
+
+    @beartype
+    def get_results(self) -> pd.DataFrame:
+        """return a list of paths to CSV files that contain the results from docking"""
+
+        file_paths = self._get_result_files("OutputFile", ".csv")
+
+        for file in file_paths:
+            from deeporigin.utils.core import fix_embedded_newlines_in_csv
 
-        files = self.parent.get_result_files_for(tool="Docking")
+            fix_embedded_newlines_in_csv(file)
 
-        chem.merge_sdf_files(files, output_sdf_file)
+        all_df = []
+        for file in file_paths:
+            df = pd.read_csv(file)
+            all_df.append(df)
+
+        df = pd.concat(all_df)
+        return df
+
+    @beartype
+    def get_poses(self) -> list[str]:
+        """return a list of paths to SDF files that contain the poses of all ligands after docking"""
+        return self._get_result_files("ResultFile", ".sdf")
+
+    def _connect(self):
+        """fetch job IDs for this protein from DB"""
+
+        jobs = self._get_jobs()
+        job_ids = [job.id for job in jobs]
+        self._make_jobs_from_ids(job_ids)
+
+    def _get_jobs(
+        self,
+        *,
+        pocket_center=None,
+        box_size=None,
+        only_with_status: Optional[list[str]] = None,
+    ) -> list:
+        """get all job IDs for this protein"""
+
+        if only_with_status is None:
+            only_with_status = ["Succeeded", "Running", "Queued", "Failed"]
+
+        filter = {
+            "status": {"$in": only_with_status},
+            "metadata": {
+                "$exists": True,
+                "$ne": None,
+            },
+            "tool": {
+                "toolManifest": {
+                    "key": "deeporigin.bulk-docking",
+                },
+            },
+        }
```

Comment:
asking the server to do this filtering can potentially save us a lot of unwanted response load